python/pyarrow/parquet/core.py

```diff
@@ -304,7 +308,15 @@ def __init__(self, source, *, metadata=None, common_metadata=None,
                  read_dictionary=None, memory_map=False, buffer_size=0,
                  pre_buffer=False, coerce_int96_timestamp_unit=None,
                  decryption_properties=None, thrift_string_size_limit=None,
-                 thrift_container_size_limit=None):
+                 thrift_container_size_limit=None, filesystem=None):
+
+        self._close_source = getattr(source, 'closed', True)
+
+        filesystem, source = _resolve_filesystem_and_path(source, filesystem)
```

Comment:
When passing a local path as string, this will now always return a filesystem object. I think that means that the `memory_map` option will be ignored (which is otherwise passed to `reader.open(..)`, and there passed to `get_reader`. 
So we might need to update `_resolve_filesystem_and_path` a bit to pass through the memory_map option (a local filesystem can take that as a param, so it can be passed through).

Reply:
Nice catch. https://github.com/apache/arrow/pull/14717/commits/570e562bd93dce69da60bb9ac4ab2591c8bfcf20